base/stats.go

```diff
@@ -1079,6 +1085,101 @@ func (s *SgwIntStat) String() string {
 	return strconv.FormatInt(s.Value(), 10)
 }
 
+// NewUIntStat creates and returns a new unsigned integer Sync Gateway stat (SgwUint64Stat).
+// The stat is initialized to initialValue and, unless SkipPrometheusStatsRegistration is true,
+// is registered with Prometheus. It returns an error if required metadata is missing or if
+// Prometheus registration fails.
+//
+// Parameters:
+//	subsystem: The Prometheus subsystem segment (e.g. resource_utilization, database) used to build the fully qualified metric name.
+//	key: The short metric key appended to the subsystem to form the final metric name.
+//	unit: The unit of measurement (e.g. bytes, seconds). Used for metadata export tooling.
+//	description: Human-readable help text for the metric. Must be non-empty.
+//	addedVersion: The Sync Gateway version the stat was introduced. Must be non-empty.
+//	deprecatedVersion: The version the stat was deprecated, or empty if not deprecated.
+//	stability: The stability level (committed, volatile, or internal). Must be non-empty.
+//	labelKeys: Slice of label keys for constant labels. Must align index-wise with labelVals.
+//	labelVals: Slice of label values corresponding to labelKeys. Length must match labelKeys.
+//	statValueType: The Prometheus value type (counter or gauge) controlling exposition semantics.
+//	initialValue: The initial uint64 value assigned to the stat's atomic counter.
+//
+// Behavior:
+//   - Validates required metadata (description, addedVersion, stability).
+//   - Builds a Prometheus descriptor with any constant labels.
+//   - Sets the initial value atomically.
+//   - Registers the metric with Prometheus unless SkipPrometheusStatsRegistration is true.
+//   - Returns (*SgwUint64Stat, error) where error is non-nil on validation or registration failure.
+//
+// Concurrency:
+//
+//	The underlying value uses atomic.Uint64 for safe concurrent mutation by callers.
+//
+// Errors:
+//
+//	Returned if required fields are missing or if prometheus.Register fails (e.g. duplicate registration).
+func NewUIntStat(subsystem, key, unit, description, addedVersion, deprecatedVersion, stability string, labelKeys, labelVals []string, statValueType prometheus.ValueType, initialValue uint64) (*SgwUint64Stat, error) {
+	stat, err := newSGWStat(subsystem, key, unit, description, addedVersion, deprecatedVersion, stability, labelKeys, labelVals, statValueType)
+	if err != nil {
+		return nil, err
+	}
+
+	wrappedStat := &SgwUint64Stat{
+		SgwStat: *stat,
+	}
+
+	wrappedStat.Set(initialValue)
+
+	if !SkipPrometheusStatsRegistration {
+		err := prometheus.Register(wrappedStat)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return wrappedStat, nil
+}
+
+func (s *SgwUint64Stat) FormatString() string {
```

Comment:
```suggestion
// FormatString returns a name of the stat type used for generating documentation.
func (s *SgwUint64Stat) FormatString() string {
```

This shows up in documentation https://docs.couchbase.com/sync-gateway/current/manage/stats-monitoring-prometheus.html and https://docs.couchbase.com/sync-gateway/current/manage/stats-monitoring-json.html

but curiously neither page use this value and it isn't generated by the generator that is used to update these pages. https://github.com/couchbase/sync_gateway/blob/da8b47516f6018f209eb2a94666bb1b87fa4313f/tools/stats-definition-exporter/stat_definition.go#L25